spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/inbound/PubSubInboundChannelAdapter.java

```diff
@@ -120,18 +119,8 @@ private void consumeMessage(ConvertedBasicAcknowledgeablePubsubMessage message)
 				this.headerMapper.toHeaders(message.getPubsubMessage().getAttributesMap());
 
 		if (this.ackMode == AckMode.MANUAL) {
-			// Send the consumer downstream so user decides on when to ack/nack.
-			messageHeaders.put(GcpPubSubHeaders.ACKNOWLEDGEMENT, new AckReplyConsumer() {
```

Comment:
We need to keep this, it's just deprecated.

Reply:
Maybe even keep a test for it.

Reply:
Good idea!